admin/sales_analytics/metrics.py

```diff
@@ -0,0 +1,113 @@
+from admin.base.settings import ENTRY_POINTS
+from admin.sales_analytics.utils import get_entry_point, get_sorted_index
+from datetime import datetime, timedelta
+from framework.mongo import database as db
+
+
+def get_user_count(db=db, entry_points=ENTRY_POINTS):
+    """
+    Get the number of users created from each entry point: osf, osf4m, prereg, and institution.
+    """
+    count_list = []
+    percent_list = []
+    tags = ['osf4m', 'prereg', 'institution', 'osf']
+    total = db.user.find({}).count()
+    for i in entry_points:
+        count = db.user.find({'system_tags': i}).count()
+        percent = round(float(count) / float(total), 2)
+        count_list.append(count)
+        percent_list.append(percent)
+    osf_count = total - sum(count_list)
+    osf_percent = 1 - sum(percent_list)
+    count_list.append(osf_count)
+    percent_list.append(osf_percent)
+    sorted_index = get_sorted_index(count_list)
+    count_list = [count_list[i] for i in sorted_index]
+    percent_list = [percent_list[i] for i in sorted_index]
+    tags = [tags[i] for i in sorted_index]
+
+    return {'tags': tags, 'count': count_list, 'percent': percent_list, 'total': total}
+
+
+def get_multi_product_metrics(db=db, timedelta=timedelta(days=365)):
+    """
+    Get the number of users using 2+ products within a period of time
+    """
+    start_date = datetime.now() - timedelta
+    pipeline = [
+        {'$match': {'date': {'$gt': start_date}}},
+        {'$group': {'_id': '$user', 'node_id': {'$addToSet': '$params.node'},
+                    'action': {'$addToSet': '$action'}}}
+    ]
+    user_node = db.nodelog.aggregate(pipeline)['result']
+    multi_product_count = 0
+    cross_product_count = 0
+    multi_action_count = 0
+    for i in user_node:
```

Comment:
Again, `i` should be reserved for numerical iterators. Also it might be good to change `user_node` to a plural. Or maybe even a name that doesn't step on the Node nomenclature of the OSF.